### DISCUSSION TITLE
Help understanding ZFS: how to add physical disks correctly with none standard configuration

### DISCUSSION THREAD

**Discussion by leuchthelp** [openzfs/zfs#16309] [General]
Hello,

I've recently been getting into zfs due to my TrueNAS Scale server and now decided to install Ubuntu with zfs on my main laptop as it seems to fit my needs best.

I got Ubuntu installed one 1x512gb WD Blue SSD for now and now want to add another 1x1tb Samsung 980 SSD to the pool. How would I best go about it with commands? TrueNAS somewhat spoiled the learning experience so far with it's GUI so I'm now trying to properly learn the concepts and right commands. 

From what I could gather the zpool attach or add commands, would be what I'm looking for. The plan is to mirror both drives and make personal files redundant and add encryption to that mirror.

Though this would mean I would lose about half of my usable space on the Samsung SSD as the mirrors have to be the same size.

Now I thought about partitioning the 1tb into 2x512, mirror one partition to the WD SSD and then have the other 512gb for other none critical storage pool like games, that can easily be reinstalled. 

How would I best go about it doing this or is it even recommended?  Appreciate the help in advance.

**Comment by amotin** [comment]
It is not a problem for ZFS to have bigger SSD to be split between several uses.  But since I suppose it is also your boot pool, I hope that loader will figure out from where to boot in that case.  And if you expect to be able to boot from the second SSD in case first fail, then you'd need to have a copy of boot partition there, may need to sync them, etc, that may be another can of worms.
